products/surveys/backend/max_tools.py

```diff
@@ -0,0 +1,196 @@
+"""
+MaxTool for AI-powered survey creation.
+"""
+
+from typing import Any
+from pydantic import BaseModel, Field
+from langchain_core.prompts import ChatPromptTemplate
+from langchain_openai import ChatOpenAI
+
+from ee.hogai.tool import MaxTool
+from posthog.models import Team
+from .prompts import SURVEY_CREATION_SYSTEM_PROMPT
+from .survey_schema import SurveyCreationOutput
+
+
+class SurveyCreatorArgs(BaseModel):
+    instructions: str = Field(description="Natural language description of the survey to create")
+
+
+class SurveyCreatorTool(MaxTool):
+    name: str = "create_survey"
+    description: str = "Create and optionally launch a survey based on natural language instructions"
+    thinking_message: str = "Creating your survey..."
+
+    root_system_prompt_template: str = """
+    You are helping create surveys for this PostHog team.
+
+    Current context:
+    - Total surveys: {total_surveys_count}
```

Comment:
Minor: we use [moustache formatting](https://github.com/noahmorrison/chevron) in the other prompts. It basically uses three parentheses instead of one : `{{{total_surveys_count}}}` and you specify "moustache" when doing `ChatPromptTemplate.from_messages`, check any node in `ee/hogai/graph` for an example.